From d6340079592e35b7b6509a22380c707ba7e3a211 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 12 May 2024 18:45:03 +1100
Subject: [PATCH 18/44] add cord_internal and string_view target for abseil 20260107

In abseil-cpp 20260107:
- libabsl_string_view.so was merged into libabsl_strings.so; add a
  :string_view alias target that depends on :strings instead.
- libabsl_cord_internal.so needs to be explicitly linked.
  Patch 0003 already adds cordz_functions/cordz_info to cord.

diff --git a/third_party/absl/system.absl.strings.BUILD b/third_party/absl/system.absl.strings.BUILD
--- a/third_party/absl/system.absl.strings.BUILD
+++ b/third_party/absl/system.absl.strings.BUILD
@@ -26,7 +26,7 @@
 
 cc_library(
     name = "cord",
-    linkopts = ["-labsl_cord", "-labsl_cordz_functions", "-labsl_cordz_info"],
+    linkopts = ["-labsl_cord", "-labsl_cordz_functions", "-labsl_cordz_info", "-labsl_cord_internal"],
     deps = [
         ":str_format",
         "//absl/container:compressed_tuple",
@@ -47,3 +47,11 @@
         "//absl/types:span",
     ],
 )
+
+cc_library(
+    name = "string_view",
+    deps = [
+        ":strings",
+        "//absl/base:nullability",
+    ],
+)
